front/src/modules/trainschedule/components/ManageTrainSchedule/Map.tsx

```diff
@@ -90,17 +90,24 @@ const Map = ({
   const infraID = useInfraID();
   const terrain3DExaggeration = useSelector(getTerrain3DExaggeration);
   const { viewport, mapSearchMarker, mapStyle, showOSM, layersSettings } = useSelector(getMap);
+  const mapRef = useRef<MapRef | null>(null);
```

Comment:
```suggestion
  const mapRef = useRef<MapRef | null>(null);
  const mapContainer = useMemo(() => mapRef.current?.getContainer(), [mapRef.current]);
```
Perhaps we could declare mapContainer once here and use it in the dependency arrays of useEffect(l.216) and useMemo(l.100) that use mapRef?